packages/core/ios/Sources/DdSdkImplementation.swift

```diff
@@ -20,18 +25,35 @@ public class DdSdkImplementation: NSObject {
     let jsDispatchQueue: DispatchQueueType
     let jsRefreshRateMonitor: RefreshRateMonitor
     let mainDispatchQueue: DispatchQueueType
-    
+    let RUMMonitorProvider: () -> RUMMonitorProtocol
+    let RUMMonitorInternalProvider: () -> RUMMonitorInternalProtocol?
+    var webviewMessageEmitter: InternalExtension<WebViewTracking>.AbstractMessageEmitter?
+
     private let jsLongTaskThresholdInSeconds: TimeInterval = 0.1;
 
     @objc
     public convenience init(bridge: RCTBridge) {
-        self.init(mainDispatchQueue: DispatchQueue.main, jsDispatchQueue: bridge, jsRefreshRateMonitor: JSRefreshRateMonitor.init())
+        self.init(
+            mainDispatchQueue: DispatchQueue.main,
+            jsDispatchQueue: bridge,
+            jsRefreshRateMonitor: JSRefreshRateMonitor.init(),
+            RUMMonitorProvider: { RUMMonitor.shared() },
+            RUMMonitorInternalProvider: { RUMMonitor.shared()._internal }
+        )
     }
     
-    init(mainDispatchQueue: DispatchQueueType, jsDispatchQueue: DispatchQueueType, jsRefreshRateMonitor: RefreshRateMonitor) {
+    init(
+        mainDispatchQueue: DispatchQueueType,
+        jsDispatchQueue: DispatchQueueType,
+        jsRefreshRateMonitor: RefreshRateMonitor,
+        RUMMonitorProvider: @escaping () -> RUMMonitorProtocol,
+        RUMMonitorInternalProvider: @escaping () -> RUMMonitorInternalProtocol?
+    ) {
         self.mainDispatchQueue = mainDispatchQueue
         self.jsDispatchQueue = jsDispatchQueue
         self.jsRefreshRateMonitor = jsRefreshRateMonitor
+        self.RUMMonitorProvider = RUMMonitorProvider
+        self.RUMMonitorInternalProvider = RUMMonitorInternalProvider
```

Comment:
Maybe worth looking into this double injections now that `._internal` isn't overly specific.

Reply:
Because `_internal` is an extension I cannot override it from a `Monitor` mock, so I need to inject it as well.

Let me know if you see a good way for doing this :)

Reply:
I had `MockRUMMonitor` extend both `RUMMonitorProtocol` and `RUMCommandSubscriber`. It's a little white-boxy, but all of the `_internal` methods send commands to the subscriber, so I check the correct commands get sent to the Subscriber.

Relevant code from the Flutter repo: https://github.com/DataDog/dd-sdk-flutter/blob/79b2c00d4b2fcc81a55093cdbea63f9e3c9d735c/packages/datadog_flutter_plugin/example/ios/Tests/DatadogRumPluginTests.swift#L498

That said, in order for `._internal` to be able to return your mock, it requires a change I made in iOS which I don't think is in a released version yet, so... this may be moot.